src/MultiTransfer.sol

```diff
@@ -0,0 +1,96 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/access/AccessControl.sol";
+
+/**
+ * @title MultiTransfer
+ * @dev This contract allows the admin to send ETH to multiple recipients in a single transaction.
+ *      If a transfer fails, it's skipped.
+ *      Failed transfers' amounts are accumulated and sent back to the admin.
+ *      Only an admin can initiate transfers and withdraw funds.
+ */
+contract MultiTransfer is AccessControl {
+    /// @notice Emitted when a transfer is successful.
+    event Transfer(address indexed to, uint256 amount);
+
+    /// @notice Emitted when a transfer fails.
+    event TransferFailed(address indexed to, uint256 amount);
+
+    /// @notice Emitted when the admin withdraws funds.
+    event Withdrawn(address indexed admin, uint256 amount);
+
+    /**
+     * @dev Sets the original `msg.sender` as the admin.
+     */
+    constructor() {
+        _setupRole(DEFAULT_ADMIN_ROLE, msg.sender);
+    }
+
+    /**
+     * @notice Transfers the specified amounts of ETH to the specified recipients.
+     * @param recipients Array of recipient addresses.
+     * @param amounts Array of amounts of ETH to send.
+     * @dev The arrays `recipients` and `amounts` must have the same length.
+     *      If a transfer fails, the transfer is skipped.
+     *      Failed transfers' amounts are accumulated and sent back to the admin.
+     *      Can only be called by the admin.
+     */
+    function multiTransfer(
+        address payable[] memory recipients,
+        uint256[] memory amounts
+    ) external onlyRole(DEFAULT_ADMIN_ROLE) {
```

Comment:
This function will be called by the IE's reward function. Does this mean we'll make the IE deploy this contract (be its admin), and also withdraw if necessary? I think this will work, and it's clean if this contract is "owned" by the IE. I'm flagging this here because you could also have meant for this to be called by human admins only

Reply:
There are two approaches here I think:
1. Having the IE deploy the contract: This would couple the IE with the rewards contract and ensures no human involvement as the IE would be the only admin. The disadvantage here is that it couples both contracts: if the reward contract requires changes, we have to redeploy the IE contract to get those changes. 
2. Deploy the Rewards contract independently: This decouples the contracts as we can reference the rewards contract by address and change it. Disadvantage here is that ultimately this requires human involvement. We can still grant the IE as an admin here and remove the human admin. 

Option no 1 is the most aligned with the IE definition in my opinion

Reply:
Cool, I'm on board with the IE deploying this contract. 

> Option no 1 is the most aligned with the IE definition in my opinion

💯  If the rewards contract can change under your feed at any time, maybe it suddenly stops working for you.

Reply:
What do you think about making `multiTransfer` payable, instead of managing a reward contract balance? Isn't that a simpler design?

Reply:
https://github.com/Meridian-IE/impact-evaluator/pull/36